src/quo/components/tags/collectible_tag/view.cljs

```diff
@@ -1,70 +1,59 @@
     [quo.components.tags.collectible-tag.style :as style]
     [quo.theme]
     [react-native.core :as rn]
     [react-native.hole-view :as hole-view]
+    [reagent.core :as reagent]
     [schema.core :as schema]))
 
-(def ?schema
-  [:=>
-   [:catn
-    [:props
-     [:map {:closed true}
-      [:options {:optional true} [:maybe [:enum false :add :hold]]]
-      [:size {:optional true} [:enum :size-24 :size-32]]
-      [:blur? {:optional true} :boolean]
-      [:theme :schema.common/theme]
-      [:collectible-img-src [:or :int :string]]
-      [:collectible-name :string]
-      [:collectible-id :string]
-      [:container-width :number]
-      [:on-layout {:optional true} [:maybe fn?]]]]]
-   :any])
-
 (defn- view-internal
   []
-  (fn [{:keys [options size blur? theme collectible-img-src collectible-name collectible-id
-               container-width on-layout]
-        :or   {size :size-24}}]
-    [rn/view
-     {:on-layout on-layout}
-     [hole-view/hole-view
-      {:holes (if options
-                [{:x            (- container-width
-                                   (case size
-                                     :size-24 10
-                                     :size-32 12
-                                     nil))
-                  :y            (case size
-                                  :size-24 -6
-                                  :size-32 -4
-                                  nil)
-                  :width        16
-                  :height       16
-                  :borderRadius 8}]
-                [])}
-      [rn/view {:style (style/container size options blur? theme)}
-       [rn/image {:style (style/collectible-img size) :source collectible-img-src}]
-       [text/text
-        {:size   :paragraph-2
-         :weight :medium
-         :style  (style/label theme)}
-        collectible-name]
-       [text/text
-        {:size        :paragraph-2
-         :weight      :medium
-         :margin-left 5
-         :style       (style/label theme)}
-        collectible-id]]]
-     (when options
-       [rn/view {:style (style/options-icon size)}
-        [icons/icon (if (= options :hold) :i/hold :i/add-token)
-         {:size     20
-          :no-color true}]])]))
+  (let [container-width (reagent/atom 0)
+        on-layout       #(->> (oops/oget % :nativeEvent :layout :width)
+                              (reset! container-width))]
+    (fn [{:keys [options blur? theme collectible-img-src collectible-name collectible-id] :as props}]
+      (let [size (or (:size props) :size-24)]
+        [rn/view
+         {:on-layout on-layout}
+         [hole-view/hole-view
+          {:holes (if options
+                    [{:x            (- @container-width
+                                       (case size
+                                         :size-24 10
+                                         :size-32 12
+                                         nil))
```

Comment:
I know you didn't make this change, but is an explicit `nil` needed ? Isn't that returned by default ?

Reply:
You are right!

Reply:
Ooh no, I'm pretty sure this is not right. Case will crash if no default value is provided so we must explicitly set it afaik

Reply:
Yes, @J-Son89. I saw it on google that `nil` is default. But according to the docs:

https://cljs.github.io/api/cljs.core/case

`If no default expression is provided and no clause matches, an Error is thrown.`

 Can I row back that?

Reply:
I think you should leave the explicit default. Afaik we've had this issue many times in the codebase 🧯

Reply:
@shivekkhurana do you also agree?